### ISSUE TITLE
Trouble with claims / priority

### ISSUE THREAD

**Issue by onpon4** [naev/naev#974]
When I put together #973 I noticed something odd, and I've come to the conclusion that the problem lies in the misn.claim() function when used with multiple systems.

These are my observations:

* If a "Seek And Destroy" mission is generated, systems it claimed seem to remain claimed forever until the game is restarted (rather than going away when the mission is destroyed like they should).
* This prevented my mission, which I set to a higher priority than "Seek And Destroy" and other computer missions, to fail to claim its system, consistently, after a "Seek And Destroy" mission was generated.
* This does not seem to affect the save files, hence why restarting the game the effect isn't carried over.
* Removing the "Seek And Destroy" mission completely eliminates the problem (note: this is the only mission that claims multiple systems, or at least the only computer mission that does so).

I don't really understand the claim system very well, but what I guess is probably happening is that the cleanup of mission claims isn't quite working right when there are more than one claimed, and something somewhere is still considering a "claimed" state to be true. Possible solutions:

* Hunt down and fix this bug
* Remove multi-system claims and modify "Seek And Destroy" to not require multi-system claims
* Remove "Seek And Destroy" mission

Duplication is hard because the exact effects are pretty non-specific and subtle. What I would suggest is creating a high-priority custom mission that requires a mission claim and prints a message when that fails, then watching it fail because of inability to claim over and over after a "Seek And Destroy" mission has been generated (which is frequent enough that usually said test mission will be locked out after a single takeoff/land on a single planet). I suspect the number of pirate bounty missions would also drop over time, but this one is far less predictable and less specific.

**Comment by nenau-again** [comment]
I'll try to look at the misn claim code, but I'm not familiar with that part of the engine either...
If the debugging takes too long, the best imo is to temporarly disable SeekNDestroy, so that the release is not impacted.

**Comment by onpon4** [comment]
So just wanted to copy what @nenau-again said on Discord onto here: that according to their tests the problem actually seemed to be that priority doesn't actually affect who gets to claim first, but rather ordering on mission.xml does (which I would consider a bug). That being the case it should be solvable temporarily just by reordering mission.xml to put important missions at the top and unimportant or computer missions at the bottom.

**Comment by onpon4** [comment]
I've included a reordering of missions in #973. This should fix the problem at least in the short-term.

**Comment by universe-11** [comment]
I finished a "Seek And Destroy" mission and not landing in that authority at once to claim it. I went on to do other missions. After a while, I want to claim it and go back there and suddenly the same target show up again. I missed to catch the target and the computer told me I did not finish the mission and had to do it again. Below is the output message... hope may help

Warning: [misn_runFunc] Mission 'Seek And Destoy' -> 'hail': [string "dat/missions/neutral/seek_n_destroy.lua"]:202: bad argument #1 to 'ipairs' (table expected, got nil)
stack traceback:
	[C]: in function 'ipairs'
	[string "dat/missions/neutral/seek_n_destroy.lua"]:202: in function 'elt_inlist'
	[string "dat/missions/neutral/seek_n_destroy.lua"]:343: in function <[string "dat/missions/neutral/seek_n_destroy.lua"]:340>
Warning: [hook_runMisn] Hook [hail] '59' -> 'hail' failed

**Comment by nenau-again** [comment]
@universe-11 : I'll look at that, but could you please open a new issue, so that if @onpon4 closes this one, yours stays open ?

**Comment by onpon4** [comment]
I think I've just fixed that in #977.

**Comment by onpon4** [comment]
I've changed the priority of this issue to Critical because ffb584bb791a690fef476a43e87b5ce66f32e5a8 means the temporary solutiton we implemented no longer works. Waiting for clarification from @bobbens regarding whether or not that change fixed this issue by causing the order of claims to be changed. If so, then this issue can be closed; if not, then this issue should be our top priority (and release-blocking) as it can lead to a lot of unique missions being completely inaccessible.

🕵️
